diff --git a/mstp.c b/mstp.c
index 7414024..8f250e4 100644
--- a/mstp.c
+++ b/mstp.c
@@ -822,9 +822,9 @@ int MSTP_IN_set_cist_bridge_config(bridge_t *br, CIST_BridgeConfig *cfg)
 
     if(cfg->set_max_hops)
     {
-        if((6 > cfg->max_hops) || (40 < cfg->max_hops))
+        if((6 > cfg->max_hops) || (255 < cfg->max_hops))
         {
-            ERROR_BRNAME(br, "Bridge Max Hops must be between 6 and 40");
+            ERROR_BRNAME(br, "Bridge Max Hops must be between 6 and 255");
             r = -1;
         }
     }
