From 2f2a85eb938bec540048f1373357d094b9b6897e Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@foss.st.com>
Date: Mon, 19 Feb 2024 18:06:05 +0100
Subject: [PATCH] Makefile_q_bridge

---
 .../ieee8021QBridgeMib-mib-module/Makefile                  | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 2e79f19..3390aba 100644
--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,11 @@
-CC=gcc
+#CC=gcc
 
-CFLAGS=-I. `net-snmp-config --cflags` -I/usr/include/base -L/usr/local/lib/libtsn/build -I/usr/include/tsn
+#CFLAGS=-I. `net-snmp-config --cflags` -I/usr/include/base -L/usr/local/lib/libtsn/build -I/usr/include/tsn
 BUILDLIBS= -lbase -ltsn
 #BUILDAGENTLIBS=`net-snmp-config --agent-libs`
 
 # shared library flags (assumes gcc)
-DLFLAGS=-fPIC -shared
+#DLFLAGS=-fPIC -shared
 
 ieee8021QBridgeMib.so: ieee8021QBridgeMib.c \
 					ieee8021QBridgeTable/ieee8021QBridgeTable.c \
-- 
2.34.1

